api/envoy/extensions/load_balancing_policies/subset/v3/subset.proto

```diff
@@ -171,6 +171,44 @@ message Subset {
   // metadata. The same host may appear in multiple subsets.
   repeated LbSubsetSelector subset_selectors = 3;
 
+  // By default, only when the request metadata has exactly **same** keys with one of subset selectors and
+  // the values of related keys are matched, the load balancer could get valid subset for the request.
+  // For example, given following subset selectors:
+  //
+  // .. code-block:: json
+  //
+  //   { "subset_selectors": [
+  //       { "keys": [ "version" ] },
+  //       { "keys": [ "stage", "version" ] }
+  //   ]}
+  //
+  // A request with metadata ``{"redundant-key": "redundant-value", "stage": "prod", "version": "v1"}`` or
+  // ``{"redundant-key": "redundant-value", "version": "v1"}`` can never get valid subset even if the values
+  // of keys ``stage`` and ``version`` are matched because of the redundant key/value pair in the request
+  // metadata.
+  //
+  // By setting this field to true, redundant key/value pairs are allowed in the request metadata. The keys
+  // of request metadata could be superset of keys of subset selectors.
+  //
+  // More specifically, if the keys of request metadata is superset of one of subset selectors, then only
+  // keys that in the selector keys will be evaluated. Take the above example, if the request metadata is
+  // ``{"redundant-key": "redundant-value", "stage": "prod", "version": "v1"}``, the load balancer would
+  // only evaluate ``stage`` and ``version`` and their values to find appropriate subset because ``stage``
+  // ``version`` are contained by the second subset selector and the redundant ``redundant-key`` will be
+  // ignored.
+  //
+  // .. note::
+  //   If the keys of request metadata is superset of multiple different subset selectors keys, the subset
+  //   selector with most keys to win. For example, given subset selectors
+  //   ``{"subset_selectors": ["keys": ["A", "B", "C"], ["A", "B"]]}`` and request metadata ``{"A": "-",
+  //   "B": "-", "C": "-", "D": "-"}``, keys ``A``, ``B``, ``C`` will be evaluated.
+  //   If the keys of request metadata is superset of multiple different subset selectors keys and the number
+  //   of selector keys are same, then the one placed in front to win. For example, given subset selectors
+  //   ``{"subset_selectors": ["keys": ["A", "B"], ["C", "D"]]}`` and request metadata ``{"A": "-", "B": "-",
+  //   "C": "-", "D": "-"}``, keys ``A``, ``B`` will be evaluated.
```

Comment:
I see how this is implemented in the code, but I didn't see a test case for it. I think it should be covered to avoid regressions since the implementation depends on the selectors being ordered longest-to-shortest but the sorting happens away from where the that precondition is used.